suite/cstest/src/test_run.c

```diff
@@ -102,17 +102,19 @@ static bool parse_input_options(const TestInput *input, cs_arch *arch,
 	}
 
 	*mode = 0;
-	bool mode_found = false;
 	size_t opt_idx = 0;
 	char **options = input->options;
 	for (size_t i = 0; i < input->options_count; ++i) {
+		bool opt_found = false;
 		opt_str = options[i];
-		val = enum_map_bin_search(test_mode_map,
-					  ARR_SIZE(test_mode_map), opt_str,
-					  &mode_found);
-		if (mode_found) {
-			*mode |= val;
-			goto next_option;
+		for (size_t k = 0; k < ARR_SIZE(test_mode_map); k++) {
```

Comment:
Please introduce the binary search again. We get a worst case number of string comparisons of `O(log2(150) = 7) * ~3 options (assuming there are on average 3 options per test case).
Vs. O(150) * 3 string comparisons.

Reply:
You can sort the table as documented in `Mapping.h::cs_enum_id_map`.

Reply:
that function is broken or similar. never worked correctly, why? dunno. that is why i have removed it.